src/components/common/NamedAddress/index.tsx

```diff
@@ -0,0 +1,21 @@
+import useAsync from '@/hooks/useAsync'
+import useChainId from '@/hooks/useChainId'
+import { getContract } from '@safe-global/safe-gateway-typescript-sdk'
+import EthHashInfo from '../EthHashInfo'
+import type { EthHashInfoProps } from '../EthHashInfo/SrcEthHashInfo'
+
+const NamedAddressInfo = ({ address, name, customAvatar, ...props }: EthHashInfoProps) => {
+  const chainId = useChainId()
+  const [contract] = useAsync(() => getContract(chainId, address), [address, chainId])
```

Comment:
Seems like this would increase the amount of requests per page quite a lot. Should we do it only if there's no name passed in props maybe?